arwen/elrondapi/manBufOps.go

```diff
@@ -424,6 +432,59 @@ func v1_4_mBufferFromBigIntSigned(context unsafe.Pointer, mBufferHandle int32, b
 	return 0
 }
 
+//export v1_4_mBufferToBigFloat
+func v1_4_mBufferToBigFloat(context unsafe.Pointer, mBufferHandle, bigFloatHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferToBigFloat
+	metering.UseGas(gasToUse)
+
+	managedBuffer, err := managedType.GetBytes(mBufferHandle)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+
+	value := managedType.GetBigFloatOrCreate(bigFloatHandle)
+
+	bigFloat := new(big.Float)
+	err = bigFloat.GobDecode(managedBuffer)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+	if oneIsInfinity(bigFloat) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+	}
+
+	value.Set(bigFloat)
+	return 0
+}
+
+//export v1_4_mBufferFromBigFloat
+func v1_4_mBufferFromBigFloat(context unsafe.Pointer, mBufferHandle, bigFloatHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferFromBigFloat
+	metering.UseGas(gasToUse)
+
+	value, err := managedType.GetBigFloat(bigFloatHandle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return 1
+	}
+
+	encodedFloat, err := value.GobEncode()
```

Comment:
Is this ok with +/- inf, or should be add a check?

Reply:
no need to check

Reply:
This would be one of the few places where infinity won't cause issues :)

Reply:
yep, anyways you absolutely shouldn't be able to get an infinity as if so, GetBigFloat() should return error and so fail execution